File: 02-back_end/spring-boot-ecommerce/src/main/java/com/hahagroup/ecommerce/dto/Purchase.java
package com.hahagroup.ecommerce.dto;

import com.hahagroup.ecommerce.entity.Address;
import com.hahagroup.ecommerce.entity.Customer;
import com.hahagroup.ecommerce.entity.Order;
import com.hahagroup.ecommerce.entity.OrderItem;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;


@Getter
@Setter
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: 03-frontend/anguler-ecommerce/src/app/common/purchase.ts
import { Customer } from './customer';
import { Address } from './address';
import { OrderItem } from './order-item';
import { Order } from './order';
export class Purchase {
    customer: Customer;
    order: Order;
    shippingAddress: Address;
    billingAddress: Address;
    orderItems: OrderItem[];

}

